proof/refine/ARM/Finalise_R.thy

```diff
@@ -2725,8 +2709,42 @@ crunch cte_wp_at'[wp]: prepareThreadDelete "cte_wp_at' P p"
 crunch valid_cap'[wp]: prepareThreadDelete "valid_cap' cap"
 crunch invs[wp]: prepareThreadDelete "invs'"
 
+crunches prepareThreadDelete
+  for sched_projs_obj_at'[wp]:
+    "\<lambda>s. obj_at' (\<lambda>tcb. tcbSchedNext tcb = None \<and> tcbSchedPrev tcb = None) t s"
```

Comment:
Very minor, but this name doesn't seem quite right to me. The following lemmas use `tcbSchedNext_tcbSchedPrev_None_obj_at'`, which is long but does feel more accurate.

Reply:
And probably not worth changing now, but do you think `tcbSchedNext tcb = None ∧ tcbSchedPrev tcb = None` could be worth an abbreviation? Or is not used that much apart from here.

Reply:
> Very minor, but this name doesn't seem quite right to me. The following lemmas use `tcbSchedNext_tcbSchedPrev_None_obj_at'`, which is long but does feel more accurate.

Ah, there's also at least one use of `tcbSchedNext_tcbSchedPrev_None` as a suffix, which is a little bit shorter. I guess I don't mind that much between the different names but being consistent would be good and can also help `crunch`.

Reply:
I think there is some inconsistency here. I'll try to clean it up. I think the `tcbSchedNext tcb = None ∧ tcbSchedPrev tcb = None` bit is used only for finalising TCBs, so I'm not sure it would make for a good abbreviation